### PR TITLE
docs: add HeaderConfig to SecurityConfig + new Gudeline Proposals

### PR DESCRIPTION
**Pull request by 39zde**
I fixed some typos. 
Also added documentation for merged PR [#11485](https://github.com/tauri-apps/tauri/pull/11485). Effective since 2.1.0

### Title
Updated title from `add HeaderConfig to SecurityConfig` to `docs: add HeaderConfig to SecurityConfig + new Gudeline Proposals`

#### New Features / Version Display
Add guideline for displaying the version, a feature was introduced via new `SinceVersion` component  

#### Badge Lifetimes
The sidebar has the ability to display badges next to an entry.
Add a guideline for how long a badge with the text contents of `New` is allowed to be displayed.

### CODE REVIEW

**Comment by FabianLars** [comment]
> Goes into effect for next minor release (v2.1.0).

Hmm, i think something like a generic `Since 2.1.0` header would be nice to have because i assume this will be a somewhat common occurence. cc @vasfvitor

**Comment by 39zde** [comment]
> > Goes into effect for next minor release (v2.1.0).
> 
> Hmm, i think something like a generic `Since 2.1.0` header would be nice to have because i assume this will be a somewhat common occurence. cc @vasfvitor

Changed it to `Effective since 2.1.0`

**Comment by vasfvitor** [comment]
> > Goes into effect for next minor release (v2.1.0).
> 
> Hmm, i think something like a generic `Since 2.1.0` header would be nice to have because i assume this will be a somewhat common occurence. cc @vasfvitor

sorry I'm not sure I get it. Do you mean that we will be having more minors with new feats that needs to be noted as `Effective since 2.x.y` at the docs?

**Comment by FabianLars** [comment]
yes, exactly. for example in the v1 docs we often mention 1.3 for the first version with nsis support or [1.1 for tauri icon](https://v1.tauri.app/v1/guides/features/icons#command-usage) but only as part of the guide text, not via a seperate visual element.

**Comment by 39zde** [comment]
This feature is included for the next minor release. This being 2.1.0. So it does not make sense to publish the documentation before that. So I changed the PR description from 
> Goes into effect for next minor release (v2.1.0).

to

> Effective since 2.1.0

If I was meant to add a notice in the documentation itself, then it's a misunderstanding on my part.

@FabianLars mentioned "[1.1 for tauri icon](https://v1.tauri.app/v1/guides/features/icons#command-usage)", which says in form of regular paragraph text:
> Starting with @tauri-apps/cli / tauri-cli version 1.1 the icon subcommand is part of the main cli:

Would adding `Effective since 2.1.0` somewhere in the guide text suffice or is `This feature is part of tauri/tauri-utils since version 2.1.0` better?

**Comment by vasfvitor** [comment]
> Effective since 2.1.0

this would be enough if on the top of the page detached from content, maybe a [banner](https://starlight.astro.build/reference/frontmatter/#banner) (if it isn't too much visually)

**Comment by 39zde** [comment]
Since there definitely will be the need for other `Since x.y.z` notices, I propose the component [SinceVersion](https://github.com/tauri-apps/tauri-docs/pull/2950/files#diff-b4661e2e35597b55e8a1985bd81fdbaefde8fc1295d07384594a0feec5bc8a92), to have something consistent going forward. 
The Comp would look like this:
dark
light
and links to the related release page.

Also I couldn't find any guidelines, on the Badge `New`, on how it should be used and how long something is considered new.
For now I just said 6 Months in [Contributing.md](https://github.com/tauri-apps/tauri-docs/pull/2950/files#diff-98e64bc1cd2db9333c6effe87bbe0d6dfe8714aba4c6bde45aa037fe0796e44c), but that's just a place holder. That's something, which requires multiple peoples input. 

On the changes to `Contributing.md` I'm not quite set, so please voice your opinions.

**Comment by 39zde** [comment]
To get [formatting Workflow](https://github.com/tauri-apps/tauri-docs/actions/runs/11805303236/job/32887484893?pr=2950) to pass, would mean breaking the page and making the code unpleasant to read. I wouldn't want to change the current formatting much, if possible. 

Could be related to #2284, but I haven't looked into it.

**Comment by FabianLars** [comment]
no problem, just add the pages that prettier would break to the .prettierignore file

**Comment by 39zde** [comment]
> Also I couldn't find any guidelines, on the Badge New, on how it should be used and how long something is considered new.
For now I just said 6 Months in [Contributing.md](https://github.com/tauri-apps/tauri-docs/pull/2950/files#diff-98e64bc1cd2db9333c6effe87bbe0d6dfe8714aba4c6bde45aa037fe0796e44c), but that's just a place holder. That's something, which requires multiple peoples input.

Say this guideline receives approval. How to enforce it? 
I have a reminder set in my calendar, but that's not solution, I suppose.

So, have a file (eg. `.github/Bage-Lifetimes.md`) with the dates of when the badge should be removed?
```markdown
# Removal Dates

List of files and dates, on which the `New` badge should be removed.
| Directive           | Description                                        |
| ------------------- | -------------------------------------------------- |
| Order               | Insert so, that the next removal is always on top. |
| Date Format         | `YYYY.MM.dd`                                       |

- [`http-headers.mdx`](../src/content/docs/security/http-headers.mdx): 2025.05.13

```

Or is there some CI workflow for notifications or something like a delayed commit. Both of these come with their own set problems and considerations.

**Comment by vasfvitor** [comment]
> Or is there some CI workflow for notifications or something like a delayed commit. Both of these come with their own set problems and considerations.

well... for now I think this is enough. It's great. I'll open an issue and let it open while we figure it out. 6 months is ok imo. 

is this pr ready?

File: src/content/docs/security/http-headers.mdx [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+content-type: text/html
+cross-origin-embedder-policy: require-corp
+cross-origin-opener-policy: same-origin
+tauri-custom-header: key1 'value1' 'value2'; key2 'value3'
+timing-allow-origin: https://developer.mozilla.org, https://example.com
+```
+
+### Frameworks
+
+Some delvelopement environments require extra settings, to emulate the production environment.
```
Comment (vasfvitor) [inline src/content/docs/security/http-headers.mdx]:
```suggestion
Some development environments require extra settings, to emulate the production environment.
```

File: src/content/docs/security/http-headers.mdx [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+          "https://example.com",
+        ],
+        "X-Content-Type-Options": null, // gets ignored
+        "Access-Control-Expose-Headers": "Tauri-Custom-Header",
+        "Tauri-Custom-Header": {
+          "key1": "'value1' 'value2'",
+          "key2": "'value3'"
+        }
+      },
+      // notice how how the CSP is not defined under headers
```
Comment (vasfvitor) [inline src/content/docs/security/http-headers.mdx]:
```suggestion
      // notice how the CSP is not defined under headers
```

**Comment by 39zde** [comment]
> is this pr ready?

I got nothing more to add. Green light from me.

File: src/content/docs/security/http-headers.mdx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        "serve":{
+          //...
+          "options":{
+            //...
+            "headers":{
+              "Cross-Origin-Opener-Policy": "same-origin",
+              "Cross-Origin-Embedder-Policy": "require-corp",
+              "Timing-Allow-Origin": "https://developer.mozilla.org, https://example.com",
+              "Access-Control-Expose-Headers": "Tauri-Custom-Header",
+              "Tauri-Custom-Header": "key1 'value1' 'value2'; key2 'value3'"
```
Comment (vasfvitor) [inline src/content/docs/security/http-headers.mdx:156]:
```suggestion
              "Tauri-Custom-Header": "key1 'value1' 'value2'; key2 'value3'"
```
is this syntax correct?. I mean, is there any missing `:` or `,`?

**Review by vasfvitor — APPROVED** [review]
thanks, great work on everything!

File: src/content/docs/security/http-headers.mdx (hunk 3 above)
Comment (39zde) [reply src/content/docs/security/http-headers.mdx:156]: not it's correct. not intuitive, but correct.

Comment (vasfvitor) [reply src/content/docs/security/http-headers.mdx:156]: ok

**Pull request merged**